### PR TITLE
Object Detection 3D Bug fixes

### PR DESCRIPTION
**Pull request by iliiliiliili**
Fixed the bug with wrong score for iterator-based evaluation and fixed the usage of absolute paths in save/load functions.

### CODE REVIEW

**Comment by stefaniapedrazzi** [comment]
I convert this PR to draft.
Please change the status when it is ready for review.

**Review by ad-daniel — CHANGES_REQUESTED** [review]
Looks good to me

**Review by ad-daniel — APPROVED** [review]
Thank you.

**Review by passalis — APPROVED** [review]
Thank you for this bugfix! I had one minor comment that puzzled me (even though it does not alter the functionality of the code).

File: src/opendr/engine/target.py [hunk 1]
```diff
@@ -415,6 +415,11 @@ def kitti(self):
         result["location"] = np.array([self.data["location"]])
         result["rotation_y"] = np.array([self.data["rotation_y"]])
         result["score"] = np.array([self.confidence])
+        num_ground_truths = 1
+        num_objects = 1
+        index = list(range(num_objects)) + [-1] * (num_ground_truths - num_objects)
```
[the remark this answers is not in the document]
Comment (iliiliiliili) [reply src/opendr/engine/target.py:640]: The Kitti format of a single box is made to be identical to the one with a list of boxes, and, I think, it's better to keep it like that for easier possible future updates.

**Pull request merged**